primefaces/src/main/java/org/primefaces/component/export/ExporterUtils.java

```diff
@@ -185,28 +183,38 @@ public static String getComponentFacetValue(FacesContext context, UIComponent pa
         return null;
     }
 
-    public static String getColumnFacetValue(FacesContext context, UIColumn column, TableExporter.ColumnType columnType) {
-        String textValue = null;
+    public static ColumnValue getColumnFacetValue(FacesContext context, UIColumn column, TableExporter.ColumnType columnType) {
+        ColumnValue columnValue = ColumnValue.EMPTY_VALUE;
         if (columnType == TableExporter.ColumnType.HEADER) {
-            textValue = Optional.ofNullable(column.getExportHeaderValue()).orElseGet(column::getHeaderText);
+            if (column.getExportHeaderValue() != null) {
+                columnValue = ColumnValue.of(column.getExportHeaderValue());
+            }
+            else {
+                columnValue = ColumnValue.of(column.getHeaderText());
+            }
         }
         else if (columnType == TableExporter.ColumnType.FOOTER) {
-            textValue =  Optional.ofNullable(column.getExportFooterValue()).orElseGet(column::getFooterText);
+            if (column.getExportFooterValue() != null) {
+                columnValue = ColumnValue.of(column.getExportFooterValue());
+            }
+            else {
+                columnValue = ColumnValue.of(column.getFooterText());
+            }
         }
 
         UIComponent facet = column.getFacet(columnType.facet());
-        if (LangUtils.isBlank(textValue) && ComponentUtils.shouldRenderFacet(facet)) {
-            textValue = getComponentValue(context, facet);
+        if (LangUtils.isBlank(columnValue.toString()) && ComponentUtils.shouldRenderFacet(facet)) {
```

Comment:
You could introduce `ColumnValue#isBlank` since ColumnValue should never be null